library/ssl_tls.c

```diff
@@ -7019,6 +7020,12 @@ static int ssl_tls12_populate_transform( mbedtls_ssl_transform *transform,
         return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
     }
 
+    ssl_mode = mbedtls_get_mode_from_ciphersuite(
+#if defined(MBEDTLS_SSL_ENCRYPT_THEN_MAC)
```

Comment:
This differs from the condition used in the signature of the enclosing function, so if for some reason someone tries to compile with a config where `MBEDTLS_ENCRYPT_THEN_MAC` is defined but `MBEDTLS_SSL_SOME_SUITES_USE_MAC` is not (which I think is not forbidden by the documentation or `check_config.h`), compilation will fail.

We should make sure we use the same condition everywhere. For the sake of simplicity/readability, perhaps it's better to use just `defined(MBEDTLS_SSL_ENCRYPT_THEN_MAC)` everywhere.

Reply:
Hmm, `MBEDTLS_SSL_SOME_SUITES_USE_MAC` is not a `mbedtls_config.h` symbol, it's derived from the value of other macros in `ssl_misc.h`. I do get a compile error with the default config minus `MBEDTLS_CIPHER_MODE_CBC` with this PR, while the same config builds fine in `development`.
```
ssl_tls.c: In function ‘mbedtls_get_mode’:
ssl_tls.c:1733:13: warning: unused parameter ‘encrypt_then_mac’ [-Wunused-parameter]
 1733 |         int encrypt_then_mac,
      |         ~~~~^~~~~~~~~~~~~~~~
ssl_tls.c: In function ‘mbedtls_get_mode_from_transform’:
ssl_tls.c:1798:22: error: ‘mbedtls_ssl_transform’ {aka ‘const struct mbedtls_ssl_transform’} has no member named ‘encrypt_then_mac’
 1798 |             transform->encrypt_then_mac,
      |                      ^~
ssl_tls.c: In function ‘ssl_tls12_populate_transform’:
ssl_tls.c:7025:41: error: ‘encrypt_then_mac’ undeclared (first use in this function)
 7025 |                                         encrypt_then_mac,
      |                                         ^~~~~~~~~~~~~~~~
ssl_tls.c:7025:41: note: each undeclared identifier is reported only once for each function it appears in
ssl_tls.c: In function ‘mbedtls_get_mode_from_transform’:
ssl_tls.c:1802:1: warning: control reaches end of non-void function [-Wreturn-type]
 1802 | }
      | ^
make[1]: *** [Makefile:289: ssl_tls.o] Error 1
make: *** [Makefile:18: lib] Error 2
```

Now that I think about it, I'm not sure my previous suggestion to use just `defined(MBEDTLS_SSL_ENCRYPT_THEN_MAC)` everywhere is the best one: alternatively, we could also define a new symbol in `ssl_misc.h`, like
```c
#if defined(MBEDTLS_SSL_SOME_SUITES_USE_CBC) && \
    defined(MBEDTLS_SSL_ENCRYPT_THEN_MAC)
#define MBEDTLS_SSL_CBC_ETM_ENABLED
#endif
```
(or a better name if you can find one) and then use that one everywhere.

Reply:
Right, indeed thanks, I'll handle this.